packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/client_config.ts

```diff
@@ -38,8 +39,10 @@ export function parseClientOptions(
     // fixes https://github.com/elastic/kibana/issues/101944
     disablePrototypePoisoningProtection: true,
     agent: {
-      maxSockets: config.maxSockets,
+      maxTotalSockets: config.maxSockets,
       keepAlive: config.keepAlive ?? true,
+      timeout: getDurationAsMs(config.idleSocketTimeout),
```

Comment:
https://nodejs.org/api/net.html#socketsettimeouttimeout-callback

Reply:
Looking at the documentation, it seems that Node itself does not close / destroy the socket.
It rather sends a `timeout` event, and then it's up to the user to end the connection.
Thus, I'm not sure the connections are being closed due to timeout ATM. I'll give it a try in local and get back to you.

Reply:
We're good, the Agent subscribes to the `'timeout'` event and destroys the socket:
https://github.com/nodejs/node/blob/main/lib/_http_agent.js#L402-L414